### PR TITLE
Update docs on serializability usage and dev check middleware

### PR DESCRIPTION
**Pull request by markerikson**
- Added more usage info on customizing the serializability middleware
- Extracted the immutability and serializability dev check middleware to their own API reference pages
- Updated Docusaurus to alpha.58
- Added subcategories to the API ref section that are expanded by default when you expand "API Reference"
- Added more details to the options TSDocs for the middleware
- Removed unwanted colors from pagination links

### CODE REVIEW

File: docs/api/configureStore.md [hunk 1]
```diff
@@ -55,7 +59,7 @@ interface ConfigureStoreOptions<S = any, A extends Action = AnyAction> {
    * If you need to customize the order of enhancers, supply a callback
    * function that will receive the original array (ie, `[applyMiddleware]`),
    * and should return a new array (such as `[applyMiddleware, offline]`).
-   * If you only need to add middleware, use the `middleware` parameter instead.
+   * If you only need to add middleware, you can use the `middleware` parameter instaead.
```
Comment (msutkowski) [inline docs/api/configureStore.md:62]:
```suggestion
   * If you only need to add middleware, you can use the `middleware` parameter instead.
```

File: docs/usage/usage-guide.md [hunk 2]
```diff
@@ -1022,17 +1022,126 @@ export const usersAdapter = createEntityAdapter({
...
+
+The [serializability dev check middleware](../api/getDefaultMiddleware.md) will automatically warn anytime it detects non-serializable values in your actions or state. We encourage you to leave this middleware active to help avoid accidentally making mistakes. However, if you _do_ need to turnoff those warnings, you can customize the middleware by configuring it to ignore specific action types, or fields in actions and state:
 
 ```js
 configureStore({
   //...
   middleware: getDefaultMiddleware({
     serializableCheck: {
-      ignoredActions: ['your/action']
+      // Ignore these actiontype
```
Comment (msutkowski) [inline docs/usage/usage-guide.md:1038]:
```suggestion
      // Ignore these action types
```

File: docs/usage/usage-guide.md [hunk 3]
```diff
@@ -1022,17 +1022,126 @@ export const usersAdapter = createEntityAdapter({
...
+
+```ts
+import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
+import {
+  getFirebase,
+  actionTypes as rrfActionTypes
+} from 'react-redux-firebase'
+import { constants as rfConstants } from 'redux-firestore'
+import rootReducer from './rootReducer'
+import { useDispatch } from 'react-redux'
```
Comment (msutkowski) [inline docs/usage/usage-guide.md:1115]:
```suggestion
```

Comment (markerikson) [reply docs/usage/usage-guide.md:1115]: Nice catches, fixing these all locally since I'm making more edits anyway.

**Pull request merged**